pkg/skaffold/runner/runcontext/v2/context.go

```diff
@@ -200,25 +207,43 @@ func (rc *RunContext) BuildConcurrency() int                         { return rc
 func (rc *RunContext) IsMultiConfig() bool                           { return rc.Pipelines.IsMultiPipeline() }
 func (rc *RunContext) GetRunID() string                              { return rc.RunID }
 
-// GetRenderOutputPath points to the directory where the manifest rendering happens. By default, it is set to "<WORKDIR>/.kpt-pipeline".
-func (rc *RunContext) GetRenderOutputPath() (string, error) {
+// GetHydrationDir points to the directory where the manifest rendering happens. By default, it is set to "<WORKDIR>/.kpt-pipeline".
+func (rc *RunContext) GetHydrationDir() (string, error) {
 	var hydratedDir string
 	var err error
-	if rc.Opts.RenderOutput == "" {
-		// Workdir is set to pwd and shall always exist.
+
+	if rc.Opts.HydrationDir == constants.DefaultHydrationDir {
 		hydratedDir = filepath.Join(rc.WorkingDir, constants.DefaultHydrationDir)
-	} else if hydratedDir, err = filepath.Abs(rc.Opts.RenderOutput); err != nil {
+	} else {
+		hydratedDir = rc.Opts.HydrationDir
+	}
+	if hydratedDir, err = filepath.Abs(hydratedDir); err != nil {
 		return "", err
 	}
+
 	if _, err := os.Stat(hydratedDir); os.IsNotExist(err) {
-		logrus.Infof("render output path does not exist, creating %v\n", hydratedDir)
+		logrus.Infof("hydrated-dir does not exist, creating %v\n", hydratedDir)
 		if err := os.MkdirAll(hydratedDir, os.ModePerm); err != nil {
 			return "", err
 		}
+	} else if !isDirEmpty(hydratedDir) {
```

Comment:
Is it possible that having existing files in this directory may interfere?  Should we be cleaning out this directory?

Reply:
The hydration dir is expected to only contain DRY/WET kubernetes manifests and a Kptfile (generated by skaffold). And users are very likely to manipulate this directory themselves. 
 
Based on this, you are right, it is possible that the existing files may interfere (what kpt in-place hydration expects to happen). And this is why we double check with customers that they are aware of this operation. 

I can't think of a better way to satisfy both conditions. Do you have any suggestions?